File: src/main/java/me/fagiolini/cinemapp/model/SearchResultModel.java
package me.fagiolini.cinemapp.model;

import io.micronaut.serde.annotation.Serdeable;
import lombok.AllArgsConstructor;
import lombok.Data;
import me.fagiolini.cinemapp.db.tables.pojos.Cinema;
import me.fagiolini.cinemapp.db.tables.pojos.Film;
import me.fagiolini.cinemapp.db.tables.pojos.Proiezione;
import me.fagiolini.cinemapp.db.tables.pojos.Sala;
@Data
@AllArgsConstructor
@Serdeable
public class SearchResultModel {
    private Film film;
    private Proiezione proiezione;
    private Cinema cinema;
    public Sala sala;
}

File: frontend/cinemapp-frontend/src/app/model/search-result-model.ts
import {Proiezione} from './proiezione';
import {Cinema} from './cinema';
import {Sala} from './sala';
import {Film} from './film';

export interface SearchResultModel {
  film: Film;
  proiezione: Proiezione;
  cinema: Cinema;
  sala: Sala;
}
